F21/mahdirahbar/lower.py

```diff
@@ -0,0 +1,155 @@
+        '''
+            This function process the words in Turkish language and 
+                convert them based on the rules in Turkish. 
+
+            input: - 
+            output:  string
+        '''
+        char_dict = {'I':'ı'}
+
+        for i in range(len(self.word_list)):            
+            for key, value in char_dict.items():
+                indexes = [i for i, ltr in enumerate(self.word_list[i]) if ltr == key]
+                for l in indexes:
+                    self.word_list[i] = self.word_list[i][:l].lower() + value + self.word_list[i][l+1:].lower()
+
+        ## An alternative way is the following two lines of code!
+
+        ## The following 2 lines of codes are a more general form of 
+        ##      converting Turkish characters to lower case.
+        # for i in range(len(self.word_list)):
+        #     self.word_list[i] = self.word_list[i].lower()
+        return self.word_to_str()
+
+    def el_lower(self):
+        '''
+            This function process the words in Greek language and 
+                convert them based on the rules in Greek. 
+
+            input: - 
+            output:  string
+        '''
+        exception_chars = {'Σ':'ς'}
+        for i in range(len(self.word_list)):
+            for key, value in exception_chars.items():
+                if key == self.word_list[i][-1]:
+                    self.word_list[i] = self.word_list[i][:-1].lower()+ value
+                else:
+                    self.word_list[i] = self.word_list[i].lower()
+
+        return self.word_to_str()
+
+
+
+    def word_to_str(self):
+        '''
+            Convert list of words to a continous string. 
+
+            input: - 
+            output: string
+        '''
+        return " ".join(self.word_list)
+
+    def utf8_decoder(self, word):
+        '''
+            Returns the decoded word.
+            input: - 
+            output: string
+        '''
+        return word.decode('utf-8')
+
+    def ga_util(self):
+        '''
+            Build the required utility for ga.
+            input: - 
+            output: None
+        '''
+        irish_upper = (u'A',u'E',u'I',u'U',u'Á',u'É',u'Í',u'Ó',u'Ú') # ,u'O'
+        irish_exp = ('n','t')
+        self.irish_upper_decoded = []
+        for i in range(len(irish_upper)):
+            self.irish_upper_decoded.append(irish_upper[i].encode('utf-8'))
+        self.irish_lower_decoded = []
+        for i in range(len(irish_exp)):
+            self.irish_lower_decoded.append(irish_exp[i].encode('utf-8'))
+
+    def call_lower(self):
+        '''
+            Calls different lower functions based on the given 
+                input language.
+
+            input: - 
+            output: returns the fianl string
+        '''
+
+        if 'en' in self.lang:
+            # English 'en'
+            return self.string.lower()
+        elif 'ga' in self.lang :
+            # Irish 'ga'
+            self.ga_util()
+            return self.ga_lower()
```

Comment:
A coding standard that I have seen in place in multiple companies is to not multiple returns in a single function. It can make it harder to trace and debug in the future. 
Rather you can assign the potential return values to a variable like `lower_string` and then return that at the end of the function.